tests/all/pooling_allocator.rs

```diff
@@ -761,21 +761,9 @@ fn dynamic_memory_pooling_allocator() -> Result<()> {
     let i32_load = instance.get_typed_func::<u32, i32>(&mut store, "i32.load")?;
     let memory = instance.get_memory(&mut store, "memory").unwrap();
 
-    // Technically this is out of bounds...
+    // This is out of bounds.
     assert!(i32_load.call(&mut store, page).is_err());
-    // ... but implementation-wise it should still be mapped memory from before.
-    // Note though that prior writes should all appear as zeros and we can't see
-    // data from the prior instance.
-    //
-    // Note that this part is only implemented on Linux which has
-    // `MADV_DONTNEED`.
     assert_eq!(memory.data_size(&store), page as usize);
-    if cfg!(target_os = "linux") {
-        unsafe {
-            let ptr = memory.data_ptr(&store);
-            assert_eq!(*ptr.offset(page as isize), 0);
-        }
-    }
```

Comment:
Could a second version of this test be executed which asserts this? One without any guard pages on dynamic memories?